src/main/java/com/github/terma/jenkins/githubprcoveragestatus/ServiceRegistry.java

```diff
@@ -8,7 +8,15 @@
     private static PullRequestRepository pullRequestRepository;
 
     public static MasterCoverageRepository getMasterCoverageRepository() {
-        return masterCoverageRepository != null ? masterCoverageRepository : Configuration.DESCRIPTOR;
+        if (masterCoverageRepository != null) {
+            return masterCoverageRepository;
+        } else {
+            if (Configuration.isUseSonarForMasterCoverage() && Configuration.getSonarUrl() != null) {
```

Comment:
Do you mean remove the condition from the if or generally remove `isUseSonarForMasterCoverage`?